core/src/main/java/tech/tablesaw/io/ReadOptions.java

```diff
@@ -327,8 +357,112 @@ public Builder minimizeColumnSizes() {
       return this;
     }
 
+    /**
+     * Provide column types for all columns preventing autodetect column type logic. It's expected
+     * that the array contains all columns
+     */
+    public Builder columnTypes(ColumnType[] columnTypes) {
+      this.columnTypes = columnTypes;
+      return this;
+    }
+
+    public Builder columnType(String columnName, ColumnType columnType) {
+      this.columnTypeMap.put(columnName, columnType);
+      return this;
+    }
+
+    /**
+     * Provide a function that determines ColumnType for some column names. To provide for all
+     * column names use {@link #completeColumnTypeByNameFunction(Function)} because it prevents
+     * running unnecessary autodetect column type logic that can be expensive in some situations
+     */
+    public Builder columnTypeByNameFunction(
+        Function<String, Optional<ColumnType>> columnTypeFunction) {
+      this.columnTypeFunction = columnTypeFunction;
+      return this;
+    }
+
+    /**
+     * Provide a function that determines ColumnType for all column names. To provide only for some
+     * use {@link #columnTypeByNameFunction(Function)}
+     *
+     * <p>Providing that function prevents running autodetect column type logic
```

Comment:
Change 
"<p>Providing that function prevents running autodetect column type logic"
to 
"<p>This method is generally more efficient because it skips column type detection."